src/coreclr/vm/jitinterface.cpp

```diff
@@ -13215,6 +13216,9 @@ static TADDR UnsafeJitFunctionWorker(
         if (g_pDebugInterface)
         {
             g_pDebugInterface->JITComplete(nativeCodeVersion, (TADDR)nativeEntry);
+#ifdef DEBUG
+            ValidateILOffsets(ftn, NULL, 0, (uint8_t*)nativeEntry, sizeOfCode);
```

Comment:
This offset validation is causing some tests to time out. Build analysis may be misclassifying some of these timeouts against unrelated known issues (e.g. https://github.com/dotnet/runtime/issues/116748)

Reply:
Agreed. I don't think I can keep this in the PR when we check in(or rather, the Validation will need to move to a less likely spot, but its been quite helpful for identifying issues.

Reply:
Instead of moving the check, I've simply adjusted the step through the method so that the total work is O(N) instead of O(N^2).  That should mitigate this problem at the expense of the check being less perfect.